### PR TITLE
Integration tests: Make audit logging inline with app log output

### PR DESCRIPTION
**Pull request by squeedee**
Make audit logging inline with app log output when running Integration tests

Now you can see the API server activity closer to the time that something happens in the controller

- also dry up the DebugControlPlane code with testEnv Creation

### CODE REVIEW

File: tests/integration/runnable/runnable_suite_test.go [hunk 1]
```diff
@@ -56,30 +56,15 @@ var (
 	controllerBuffer *gbytes.Buffer
 )
 
-const DebugControlPlane = false
+const DebugControlPlane = true
```
Comment (waciumawanjohi) [inline tests/integration/runnable/runnable_suite_test.go]: This looks like it should be removed.

File: tests/integration/supplychain/supply_chain_suite_test.go [hunk 2]
```diff
@@ -65,23 +65,7 @@ var _ = BeforeSuite(func() {
...
-
-		AttachControlPlaneOutput: DebugControlPlane, // Set to true for great debug logging
-	}
-
-	if DebugControlPlane {
-		testEnv.ControlPlane.GetAPIServer().Configure().
-			Append("audit-policy-file", filepath.Join(workingDir, "policy.yaml")).
-			Append("audit-log-path", "-")
-	}
+	testEnv = integration.CreateTestEnv(workingDir, GinkgoWriter)
```
Comment (waciumawanjohi) [inline tests/integration/supplychain/supply_chain_suite_test.go:66]: The `const DebugControlPlane = false` 8 lines prior looks like it should be removed.

File: tests/integration/runnable/runnable_suite_test.go (hunk 1 above)
Comment (squeedee) [reply tests/integration/runnable/runnable_suite_test.go]: Oops good catch

**Comment by waciumawanjohi** [comment]
The de-duplication makes sense.

I tried running the integration tests on main and on this branch, inducing a failure in each. I didn't see a difference in behavior. Maybe I wasn't creating the right type of mess? Can you suggest steps I can take to observe the change?

**Comment by squeedee** [comment]
> The de-duplication makes sense.
> 
> I tried running the integration tests on main and on this branch, inducing a failure in each. I didn't see a difference in behavior. Maybe I wasn't creating the right type of mess? Can you suggest steps I can take to observe the change?

@waciumawanjohi 

On my demo branch ([demo-audit-logs](https://github.com/vmware-tanzu/cartographer/tree/demo-audit-logs)):
- I set the constant `DebugControlPlane` to True.
- I added a fail at the end of a runnable integration test
- Observe that audit logs (`"apiVersion":"audit.k8s.io/v1"`) are interpersed between our logs (`"logger":"controller.runnable-service"`):
  ```
  {"level":"info","ts":1638983136.663757,"logger":"controller.runnable-service", ...
  {"kind":"Event","apiVersion":"audit.k8s.io/v1","level":"Request","auditID":"36e6 ...
  {"kind":"Event","apiVersion":"audit.k8s.io/v1","level":"Request","auditID":"36e6 ... 
  {"level":"info","ts":1638983136.667907,"logger":"controller.runnable-service", ...
  ```

On the "main demo branch" (https://github.com/vmware-tanzu/cartographer/tree/old-audit-log-demo):
- I set the DebugControlPlane to True
- I added the same fail in the runnable integration test
- observe that the audit logs are dumped before the controller logs (and/or after) but not interspersed

**Comment by squeedee** [comment]
I'll just add, that by interspersing the logs, @idoru and I were able to debug reconciler behavior accurately because CRUD operations happened in sequence with our controller logging.

File: tests/integration/test_helpers.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+			filepath.Join(workingDir, "..", "..", "..", "config", "crd", "bases"),
+			filepath.Join(workingDir, "..", "..", "resources", "crds"),
+		},
+		AttachControlPlaneOutput: DebugControlPlane, // Set to true for great debug logging
+	}
+
+	if DebugControlPlane {
+		apiServer := testEnv.ControlPlane.GetAPIServer()
+		apiServer.Out = out
+		apiServer.Err = out
```
Comment (squeedee) [inline tests/integration/test_helpers.go:41]: This is the change that actually "makes audit logging inline"

**Review by waciumawanjohi — APPROVED** [review]

**Pull request merged**